src/dockerhub.js

```diff
@@ -0,0 +1,37 @@
+var rp = require( "request-promise" );
+var urlencode = require( "urlencode" );
+var when = require( "when" );
+
+function onRequest( body ) {
+	return body;
+}
+
+function onError( err ) {
+	console.error( "There was an error checking a tags existance on docker hub, ", err );
+	return [];
+}
+
+function listTags( namesapce, name ) {
+	var options = {
+		uri: "https://registry.hub.docker.com/v1/repositories/" + urlencode( namesapce ) + "/" + urlencode( name ) + "/tags",
+		json: true,
+		headers: {
+			Authorization: "Basic " + new Buffer( process.env.DOCKER_USER + ":" + process.env.DOCKER_PASS ).toString( "base64" )
+		}
+	};
+	return rp( options ).then( onRequest ).catch( onError );
+}
+
+function checkExistance( namesapce, name, tag ) {
+	return listTags( namesapce, name ).then( function( tags ) {
+		var arrayFound = tags.filter( function( item ) {
```

Comment:
I think we should use something faster than a linear search; I don't think there is an upper bound on the number of tags a docker image can have. I bet that there is something in lodash or some other npm module that can do this (e.g. binary search, etc)